src/index.ts

```diff
@@ -1,18 +1,43 @@
+import axios, {AxiosError, AxiosRequestConfig, AxiosResponse} from 'axios';
 import * as extend from 'extend';
-import * as _request from 'request';  // for types only
-import * as request from 'retry-request';
+import * as rax from 'retry-axios';
 
 const BASE_URL = 'http://metadata.google.internal/computeMetadata/v1';
 
-export type Options = _request.CoreOptions&{property?: string, uri?: string};
+export type Options = AxiosRequestConfig&{property?: string, uri?: string};
 
 export type Callback =
-    (error: NodeJS.ErrnoException|null, response?: _request.RequestResponse,
+    (error: NodeJS.ErrnoException|null, response?: AxiosResponse<string>,
      metadataProp?: string) => void;
 
+// Accepts an options object passed from the user to the API.  In the
+// previous version of the API, it referred to a `Request` options object.
+// Now it refers to an Axios Request Config object.  This is here to help
+// ensure users don't pass invalid options when they upgrade from 0.4 to 0.5.
+// tslint:disable-next-line no-any
+function validate(options: any, callback: (err?: Error) => void) {
```

Comment:
For synchronous functions, prefer not to use callbacks. This will also address the nits I have added below.

Reply:
I misunderstood the previous comment asking for consistent style here.  Fixed 👍